api/user.go

```diff
@@ -1159,7 +1159,7 @@ func completeSaml(c *Context, w http.ResponseWriter, r *http.Request) {
 		}
 		return
 	} else {
-		if err := c.App.CheckUserAdditionalAuthenticationCriteria(user, ""); err != nil {
+		if err := c.App.CheckUserAllAuthenticationCriteria(user, ""); err != nil {
```

Comment:
This flow was always a bit weird in that the "password" part of the authentication is unilaterally tested first, then the other authentication schemes. I don't think there's a way around that until the LDAP server returns the username by which we can run the remaining checks, so I'm not doing anything here.